db/main.go

```diff
@@ -0,0 +1,191 @@
+	if err != nil {
+		log.Fatalf("falha ao criar cliente do Google Drive, erro %q", err)
+	}
+	if err := summarize(*source, *state, datastoreClient, googleDriveService); err != nil {
+		log.Fatalf("falha ao executar processamento do resumidor do banco, erro %q", err)
+	}
+}
+
+func createGoogleDriveClient(googleDriveCredentialsFile, goodleDriveOAuthTokenFile string) (*drive.Service, error) {
+	b, err := ioutil.ReadFile(googleDriveCredentialsFile)
+	if err != nil {
+		log.Fatalf("falha ao ler arquivo de crendenciais [%s], erro %q", googleDriveCredentialsFile, err)
+	}
+	config, err := google.ConfigFromJSON(b, drive.DriveScope)
+	if err != nil {
+		return nil, fmt.Errorf("falha ao processar configuraçōes usando o arquivo [%s], erro %q", googleDriveCredentialsFile, err)
+	}
+	f, err := os.Open(goodleDriveOAuthTokenFile)
+	if err != nil {
+		return nil, fmt.Errorf("falha ao abrir arquivo de token oauth [%s], erro %q", goodleDriveOAuthTokenFile, err)
+	}
+	defer f.Close()
+	tok := &oauth2.Token{}
+	if err = json.NewDecoder(f).Decode(tok); err != nil {
+		return nil, fmt.Errorf("falha ao fazer bind do token OAuth, erro %q", err)
+	}
+	googleDriveService, err := drive.New(config.Client(context.Background(), tok))
+	if err != nil {
+		return nil, fmt.Errorf("não foi possível criar Google Drive service, erro %q", err)
+	}
+	return googleDriveService, nil
+}
+
+func summarize(source, state string, datastoreClient *datastore.Client, googleDriveService *drive.Service) error {
+	query := fmt.Sprintf("name contains '%s' and '%s' in parents", state, source) // pegando os arquivos com prefixo 'estado' da pasta de id 'source'
+	fileList, err := googleDriveService.Files.List().Q(query).Do()
+	if err != nil {
+		return fmt.Errorf("falha ao buscar arquivos do estado [%s] no diretório [%s], erro %q", state, source, err)
+	}
+	candFiles := getCandidateFiles(fileList)
+	dbItems, err := getDBItems(candFiles, googleDriveService)
+	if err != nil {
+		return fmt.Errorf("falha ao gerar itens do banco, erro %q", err)
+	}
+	for _, c := range dbItems {
+		userKey := datastore.NameKey(candidaturesCollection, fmt.Sprintf("%s_%s", c.State, c.City), nil)
+		if _, err := datastoreClient.Put(context.Background(), userKey, c); err != nil {
+			return fmt.Errorf("falha ao salvar cidade [%s] do estado [%s] no banco, erro %q", c.City, c.State, err)
+		}
+		log.Printf("saved city [%s] of state [%s]", c.City, c.State)
+	}
+	return nil
+}
+
+func getCandidateFiles(fileList *drive.FileList) map[string]*gDriveCandFiles {
+	candFiles := make(map[string]*gDriveCandFiles)
+	for _, item := range fileList.Files {
+		sequencialID := re.FindAllString(item.Name, -1)[0]
+		switch filepath.Ext(item.Name) {
+		case ".pb":
+			found := candFiles[sequencialID]
+			if found == nil {
+				candFiles[sequencialID] = &gDriveCandFiles{
+					candidatureFile: item,
+				}
+			} else {
+				found.candidatureFile = item
+			}
```

Comment:
Ao fazer a mudança que falei acima, esse código fica muito mais simples. Uma vez que o mapa, quando não encontra, retorna o valor zero da struct.
```suggestion
			candFiles[sequencialID].candidatureFile = item
```

Reply:
Eu tentei essa implementação inicialmente, mas a IDE acusava esse erro abaixo. Tem alguma sugestão do que pode ser isso? 
![image](https://user-images.githubusercontent.com/17463488/91507924-7bf69d00-e8ac-11ea-905d-d2d0f74e09ab.png)

Reply:
Aplica a mudança e atualiza o pr para eu conseguir analisar. Você fez uma busca sobre o erro?

Reply:
Fiz uma busca sim (https://blog.ijun.org/2017/07/cannot-assign-to-struct-field-in-map.html) .Parece ser um problema com os endereços dos campos do mapa. Vou atualizar o PR com o código que temos erro